docs/recipes/ImplementingUndoHistory.md

```diff
@@ -0,0 +1,140 @@
+# Implementing Undo History
+
+[redux-undo](https://github.com/omnidan/redux-undo) is a library that provides simple undo/redo functionality for any part of your redux tree.
+
+In this recipe, you will learn how to make the [Todo List example](http://rackt.github.io/redux/docs/basics/ExampleTodoList.html) undoable - all it takes are [two lines of code](https://twitter.com/dan_abramov/status/647040825826918400) and a few terminal commands!
+
+You can find the full source of this recipe in [the redux examples folder](https://github.com/rackt/redux/tree/master/examples/todos-with-undo).
+
+
+## Installing redux-undo
```

Comment:
I think that, just before this section, we need a section *explaining* how `redux-undo` works, without mentioning it by name. It doesn't need to be long—we just need to introduce the idea of a higher-order reducer, and how one would go about implementing a trivial `undoable` themselves.